htlcswitch/interfaces.go

```diff
@@ -208,8 +208,18 @@ const (
 
 // OptionalBandwidth is a type alias for the result of a bandwidth query that
 // may return a bandwidth value or fn.None if the bandwidth is not available or
-// not applicable.
-type OptionalBandwidth = fn.Option[lnwire.MilliSatoshi]
+// not applicable. IsHandled is set to false if the external traffic shaper does
+// not handle the channel in question.
+type OptionalBandwidth struct {
+	// IsHandled is true if the external traffic shaper handles the channel.
+	// If this is false, then the bandwidth value is not applicable.
+	IsHandled bool
```

Comment:
when does this set to true?

Reply:
Argh, good catch. Copy/paste mistake. Fixed!
We'll validate all of this with an integration test in `litd`. Unfortunately we aren't really set up to test these things properly in unit or integration tests with in `lnd` (but probably worth the investment very soon).